src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int count;
+    public static final int MAX_CAPACITY = 10;
+    private Object[] elements = new Object[MAX_CAPACITY];
     @Override
     public void add(T value) {
+        if (count >= elements.length) {
+            elements = grow();
+        }
+        elements[count] = value;
+        count++;
 
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexforAdd(index);
+        count++;
+        Object[] newArray = new Object[elements.length + 1];
+        if (count > MAX_CAPACITY) {
+             newArray = grow();
+
+        }
+        for (int i = 0; i < newArray.length; i++) {
+            if (i == index) {
+                break;
+            }
+            newArray[i] = elements[i];
+        }
+        newArray[index] = value;
+        for (int i = index + 1; i < newArray.length; i++) {
+            newArray[i] = elements[i - 1];
+        }
+        elements = newArray;
 
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (count > elements.length) {
+            elements = grow();
+        }
+        for (int i = 0; i < elements.length; i++) {
+            elements[i] = list.get(i);
+        }
 
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexforAdd(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T element = (T) elements[index];
+        Object[] newArray1 = new Object[elements.length - 1];
```

Comment:
Creating a new array with `elements.length - 1` is incorrect. The array should remain the same size, and elements should be shifted to fill the gap left by the removed element.